pkgs/applications/audio/stargate/default.nix

```
@@ -0,0 +1,169 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, substituteAll
+, alsa-lib
+, autoconf
+, automake
+, ffmpegSupport ? true
+, ffmpeg
+, fftw
+, fftwSinglePrec
+, file
+, gdb
+, gettext
+, git
+, jq
+, lameSupport ? true
+, lame
+, libsbsms
+, libsndfile
+, pkg-config
+, portaudio
+, portmidi
+, python3
+, python3Packages
+, rubberband
+, stargate-libcds
+, vorbisToolsSupport ? true
+, vorbis-tools
+}:
+
+assert ffmpegSupport -> ffmpeg != null;
+assert lameSupport -> lame != null;
+assert vorbisToolsSupport -> vorbis-tools != null;
+
+let
+  python3CommonInputs = python3.withPackages
+    (ps: with ps; [
+      jinja2
+      mido
+      mutagen
+      numpy
+      psutil
+      pymarshal
+      pyqt5
+      pyyaml
+      wavefile
+      wheel
+    ]);
+in
+stdenv.mkDerivation rec {
+  pname = "stargate";
+  version = "21.11.5";
+  disabled = python3Packages.pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "stargateaudio";
+    repo = pname;
+    rev = "release-${version}";
+    sha256 = "sha256-d5ZuDCyq9gvfZvq57XHKEq0pvi+dWgz6z34kFIOmj7A=";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    gdb
+    git
+    jq
+    pkg-config
+    python3Packages.cython
+    python3Packages.setuptools
+  ];
+
+  buildInputs = [
+    fftw
+    fftwSinglePrec
+    gettext
+    libsbsms
+    libsndfile
+    portaudio
+    portmidi
+    python3
+    rubberband
+    stargate-libcds
+    file
+  ]
+  ++ lib.lists.optional stdenv.isLinux alsa-lib
+  ++ lib.lists.optional ffmpegSupport ffmpeg
+  ++ lib.lists.optional lameSupport lame
+  ++ lib.lists.optional vorbisToolsSupport vorbis-tools;
+
+  propagatedBuildInputs = [
+    python3CommonInputs
+  ];
+
+  checkInputs = [
+    python3CommonInputs
+    python3Packages.pytest
+    python3Packages.pytest-cov
+    python3Packages.yq
+  ];
```

```suggestion
  checkInputs = with python3Packages; [
    python3CommonInputs
    pytest
    yq
  ];
```

We can reduce the duplication of python3Packages.
Generating coverage is not useful, we can't send it anywhere and it makes the tests slower.